icu4c/source/i18n/calendar.cpp

```diff
@@ -1610,8 +1584,19 @@ void Calendar::computeWeekFields(UErrorCode &ec) {
     if(U_FAILURE(ec)) {
         return;
     }
+
+    // Compute day of week: JD 0 = Monday
+    int32_t dayOfWeek = julianDayToDayOfWeek(fFields[UCAL_JULIAN_DAY]);
+    internalSet(UCAL_DAY_OF_WEEK, dayOfWeek);
+    int32_t firstDayOfWeek = getFirstDayOfWeek();
+    // Calculate 1-based localized day of week
+    int32_t dowLocal = dayOfWeek - firstDayOfWeek + 1;
+    if (dowLocal < 1) {
+        dowLocal += 7;
+    }
+    internalSet(UCAL_DOW_LOCAL,dowLocal);
```

Comment:
After this, the old code also did
```
    fFields[UCAL_DOW_LOCAL] = dowLocal;
```
Is that no longer necessary?

Reply:
I remove that line because the internalSet(UCAL_DOW_LOCAL, dowLocal) in the previous line did that already
See 
```
inline void
Calendar::internalSet(UCalendarDateFields field, int32_t value)
{
    fFields[field] = value;
    fStamp[field] = kInternallySet;
    fIsSet[field]     = true; // Remove later
}
```